Addons/tsgExoticQuadrature.hpp

```diff
@@ -178,28 +197,77 @@ inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double sh
                                     std::move(weights_cache), description);
 }
 
-// Basic function overloads.
-
-// If only the weight function is provided, the reference points and weights are generated by the GL rule and the weight_fn values
-// are generated by applying the weight function to these reference points.
+/*
+ * \ingroup TasmanianAddons
+ * \brief Constructs a CustomTabulated object containing the exotic quadrature nodes and quadrature weights for a given level \b n,
+ * scalar \b shift, a reference TasmanianSparseGrid \b grid containing loaded values of the weight function, a string \b description,
+ * and an optional bool \b is_symmetric that should be set to true if the weight function is symmetric.
+ *
+ * It is assumed that grid has the following properties:
+ *     (i)   grid.getPoints() returns the quadrature reference points used to compute the entries of the Jacobi matrix.
```

Comment:
(i) and (iii) combined means that you have a sparse grid interpolant/surrogate to the weight function. It is cleaner to say that you want an interpolant to the weight function, but we will only use the values at the points and the associated quadrature weights.

Also, the user wouldn't know what Jacobi matrix means. There are several Jacobi matrices in math, it is better to say that the point will be used to orthogonalize the polynomial basis needed to compute the optimal exotic quadrature.

Reply:
I changed the descriptions according to your suggestions.